packages/cli/src/lib/implementation/global.utils.ts

```diff
@@ -32,6 +34,11 @@ export function logErrorBeforeThrow<T extends (...args: any[]) => any>(
       // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-argument
       return await fn(...args);
     } catch (error) {
+      if (error instanceof OptionValidationError) {
+        ui().logger.error(error.message);
+        await new Promise(resolve => process.stdout.write('', resolve));
+        yargs().exit(1, error);
```

Comment:
Is this to flush the output stream before exiting? Might be worth a comment to explain.

Reply:
Agree, the comment is updated 👌 

Also, since there is an open issue in the `yargs` repository regarding `yargs().exit()` suppressing errors, I added some tests for `logErrorBeforeThrow`. They make sure that when `OptionValidationError` is thrown, the custom error is logged and `yargs` exits properly internally using `process.exit(1)` without rethrowing the error. Could you please take a look? It would be great to have reliable coverage to detect regressions if `yargs` logic changes in the future.